design/proposals/variable_provision.md

```diff
@@ -0,0 +1,295 @@
+# Solution Design - Variable Provisioning
+
+## Table of Contents
+
+- [Glossary](#glossary)
+- [Useful links](#useful-links)
+- [Background](#background)
+- [Issue description](#issue-description)
+  * ['Browsing-up` for secrets management](#-browsing-up--for-secrets-management)
+  * [Secrets Domain Knowledge Required](#secrets-domain-knowledge-required)
+  * [Human Intervention Required](#human-intervention-required)
+  * [Credential Provenance](#credential-provenance)
+- [Solution](#solution)
+  * [Design](#design)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Version update](#version-update)
+- [Open questions](#open-questions)
+- [Breaking](#breaking)
+
+
+## Glossary
```

Comment:
Can we lead with the problem to grab the reader? Maybe something like:
```md
## Problem Definition
Conjur currently suffers a couple of shortcomings related to the creation of variable values:
1. `Browsing-up`: secret values are generated on a less secure platform to load into a more secure platform (Conjur)
1. Domain Knowledge: generating certificates require a deep understanding which increases the likelihood of incorrect configuration
1. Provenance: we currently lack visibility into the creation of a variable. 
...
```

Reply:
Are you suggesting moving the `Issue description` to the top and renaming it? Or adding a `Problem Definition` that summarizes the issue(s)?

Reply:
Kind of both 😛 . I like the idea of starting a proposal like this by defining the problem so we can first establish this is a real problem that substantially impacts customers.  Let's move the `Issue Description` section up. 

As a side note, I don't feel like I have a good vocabulary for what to call the problem definition ☹️